ludwig/datasets/mixins/kaggle.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+from kaggle.api.kaggle_api_extended import KaggleApi
+
+DEFAULT_CACHE_LOCATION = str(Path.home().joinpath('.ludwig_cache'))
+api = KaggleApi()
+api.authenticate()
+
+
+class KaggleMixin:
+    """A mixin to abstract away the details of the kaggle API which includes
+    the ability to authenticate against the kaggle API, list the various datasets
+    and finally download the dataset"""
+    config: dict
+    raw_dataset_path: str
+    raw_temp_path: str
+    name: str
```

Comment:
Here we can assume the presence of member variables for the username as key:

```
kaggle_username: str
kaggle_api_key: str
```

Reply:
Added